app/models/authentication.rb

```diff
@@ -152,20 +151,16 @@ def native_ref
 
   private
 
-  def set_credentials_changed_on
-    return unless @auth_changed
-    self.credentials_changed_on = Time.now.utc
-  end
-
   def after_authentication_changed
-    return unless @auth_changed
+    return unless saved_change_to_userid? || saved_change_to_password? || saved_change_to_auth_key?
```

Comment:
I don't think this is the right way to do this. This means that if we ever change what attributes constitute an important change we would need to know to change this line.

That doesn't seem maintainable to me.